### ISSUE TITLE
Custom schema is ignored if @RequestBody is referenced

### ISSUE THREAD

**Issue by neon-dev** [smallrye/smallrye-open-api#116]
If I write this:
```java
@POST
public void test(@RequestBody(content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = DifferentObject.class)), required = true) SomeObject body) { ... }
```
then everything is as expected:
```yml
    post:
      requestBody:
        $ref: '#/components/requestBodies/test'
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/DifferentObject'
```

But if I use a `ref` instead:
```java
@POST
public void test(@RequestBody(ref = "test") SomeObject body) { ... }
```
```java
@OpenAPIDefinition(
    components = @Components(
        requestBodies = @RequestBody(name = "test", content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = DifferentObject.class)), required = true)
    )
)
```
then the schema ref is incorrect:
```yml
    post:
      requestBody:
        $ref: '#/components/requestBodies/test'
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/SomeObject'
```

**Comment by MikeEdgar** [comment]
@neon-dev , in your example with the implementation specified in the `@RequestBody` the result shows a generated `$ref` to `#/components/requestBodies/test`. Can you confirm that behavior?

**Comment by neon-dev** [comment]
You mean `ref`, not implementation, right? So the second example, yes.
I just tried it again and got the following:
First example (directly specified `@RequestBody`):
```yml
  /rest/test:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/DifferentObject'
        required: true
      responses:
        201:
          description: Created
```
Second example (`@RequestBody` using `ref`):
```yml
  /rest/test:
    post:
      requestBody:
        $ref: '#/components/requestBodies/test'
        content:
          '*/*':
            schema:
              $ref: '#/components/schemas/SomeObject'
      responses:
        201:
          description: Created
```

Also note that the second one says content type `*/*` instead of `application/json`. This changes to `application/json` after you annotate the method with `@Consumes(MediaType.APPLICATION_JSON`

**Comment by MikeEdgar** [comment]
Ok, that's what I would expect, thanks. The very first YAML output example had a `$ref` that threw me off.

**Comment by neon-dev** [comment]
Yeah that was my bad, sorry. I guess I had the ref still defined in `@OpenAPIDefinition` or something like that. Was testing back and forth before finding the issue and creating this report.

**Comment by MikeEdgar** [comment]
@neon-dev - FYI #118 should help, if accepted.

**Comment by neon-dev** [comment]
Looks good 👍
Found a typo in your [code comment ](https://github.com/smallrye/smallrye-open-api/pull/118/files#diff-c9faacca95afa9ce31f5830290b39b6cR613) ("is" must be "if")

**Comment by MikeEdgar** [comment]
Fixed by #118. I missed adding the "fixes" reference in the PR.
